components/FilterByCategory.js

```diff
@@ -0,0 +1,187 @@
+            }}
+          >
+            <BackgroundIcons>
+              <IconWater />
+            </BackgroundIcons>
+            <LabelIcons>Water</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Mountains")}
+            style={{
+              color: selectedByIcon.includes("Mountains") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconMountains />
+            </BackgroundIcons>
+            <LabelIcons>Mountains</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Winter")}
+            style={{
+              color: selectedByIcon.includes("Winter") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconWinter />
+            </BackgroundIcons>
+            <LabelIcons>Winter</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Biking")}
+            style={{
+              color: selectedByIcon.includes("Biking") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconBiking />
+            </BackgroundIcons>
+            <LabelIcons>Biking</LabelIcons>
+          </ContainerIconLabel>
+        </Row>
+
+        <Row>
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Hiking")}
+            style={{
+              color: selectedByIcon.includes("Hiking") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconHiking />
+            </BackgroundIcons>
+            <LabelIcons>Hiking</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Extrem")}
+            style={{
+              color: selectedByIcon.includes("Extrem") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconExtreme />
+            </BackgroundIcons>
+            <LabelIcons>Extreme</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Indoor")}
+            style={{
+              color: selectedByIcon.includes("Indoor") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconIndoor />
+            </BackgroundIcons>
+            <LabelIcons>Indoor</LabelIcons>
+          </ContainerIconLabel>
+
+          <ContainerIconLabel
+            onClick={() => handleIconClick("Others")}
+            style={{
+              color: selectedByIcon.includes("Others") ? "orange" : "black",
+            }}
+          >
+            <BackgroundIcons>
+              <IconOthers />
+            </BackgroundIcons>
+            <LabelIcons>Others</LabelIcons>
+          </ContainerIconLabel>
+        </Row>
```

Comment:
This is a lot of repetitive code. Can you find a solution where you map over a single array of categories? Hint: you would need to get rid of your "Row" component and display the elements in a grid or wrapping flexbox.

Reply:
✅